File: TellDontAsk/AlarmsObjectOriented/Criterias/WeekendCriteria.cs
using System;
using AlarmsObjectOriented.Interfaces;
using TddEbook.TddToolkit;

namespace AlarmsObjectOriented.Criterias
{
  internal class WeekendCriteria : TimeCriteria
  {
    public bool IsSatisfied()
    {
      return Any.Boolean();
    }

    public void Output()
    {
      Console.WriteLine("it's weekend");
    }
  }
}

File: Java/AlarmsObjectOriented/src/main/java/criterias/WeekendCriteria.java
package criterias;

import interfaces.TimeCriteria;

import static autofixture.publicinterface.Generate.any;

public class WeekendCriteria implements TimeCriteria {
  public boolean isSatisfied() {
    return any(boolean.class);
  }

  public void output() {
    System.out.println("it's weekend");
  }
}
